idaes/power_generation/carbon_capture/compression_system/compressor.py

```diff
@@ -0,0 +1,411 @@
+        # Need to extend for mixture
+        # calculate total enthalpy and entropy change through the stage
+         
+
+        @self.Expression(self.flowsheet().config.time, doc="Specific enthalpy change of isentropic process")
+        def delta_enth_isentropic(b,t):
+            return b.properties_isentropic[t].enth_mol - properties_in[t].enth_mol  
+
+        @self.Expression(self.flowsheet().config.time, doc="Actual enthalpy change")
+        def delta_enth_actual(b,t):
+            return properties_out[t].enth_mol - properties_in[t].enth_mol 
+
+        @self.Expression(self.flowsheet().config.time, doc="Entropy change in compressor stage")
+        def deltaS(b,t):
+            return properties_out[t].entr_mol - properties_in[t].entr_mol   
+
+        @self.Constraint(self.flowsheet().config.time, doc="Isentropic efficiency")
+        def eff_isen_eqn(b,t):   
+            eff = b.efficiency_isentropic[t]     
+            return b.delta_enth_isentropic[t] == eff * b.delta_enth_actual[t]      
+
+        @self.Constraint(self.flowsheet().config.time)
+        def deltaH_t_con(b,t):
+            mw = properties_in[t].mw
+            return b.delta_enth_actual[t] == b.I[t] * b.U2[t]**2 * mw
+
+        @self.Constraint(self.flowsheet().config.time)
+        def deltaH_p_con(b,t):
+            mw = properties_in[t].mw
+            return b.delta_enth_polytropic[t] == b.mu_p[t] * self.U2[t]**2 * mw      
+
+        # This equation of polytropic enthalpy is based on Eq. (2.13) From Aungier textbook (2000)
+        @self.Expression(self.flowsheet().config.time, doc="Equation for polytropic enthalpy") 
+        def delta_enth_polytropic_con(b,t):
+            Tout = properties_out[t].temperature
+            Tin = properties_in[t].temperature
+            return b.delta_enth_polytropic[t] == b.delta_enth_actual[t]-b.deltaS[t]*(Tout-Tin)/log(Tout/Tin)     
+        
+        
+        # -------------------------------------------------------------------------------------------------------
+        # design variable constraints
+        @self.Constraint(self.flowsheet().config.time, doc="Maximum allowable impeller tip speed") 
+        def U2max_cons(b,t):
+            phi = b.mass_flow_coeff[t]
+            return b.U2[t] <= sqrt((1984.1*phi**2 - 616.88*phi + 215.97)*0.7*830)
+
+        # @self.Constraint(self.flowsheet().config.time, doc='Maximum Ma number')
+        # def Mamax_cons(b,t):
+        #     # n_stage = 8
+        #     return b.Ma[t] <= -0.202*log(5) + 1.25
+
+        @self.Expression(self.flowsheet().config.time, doc="Thermodynamic power")
+        def fluid_pow(b,t):
+            flow = properties_in[t].flow_mol 
+            return b.delta_enth_actual[t]*flow
+
+        @self.Expression(self.flowsheet().config.time, doc="Shaft power")
+        def elec_pow(b,t):
+            return b.fluid_pow[t]/(b.efficiency_mech * b.eff_drive)
+
+    def initialize(self, state_args={}, outlvl=6, solver='ipopt',
```

Comment:
Outlvl and reporting should use the idaes logger and the predefined levels.

Reply:
should be ```outlvl=idaeslog.NOTSET```.  That will use the default logger settings.  The levels are an enumerated type.  6 won't cause an error, but it will be roughly the same a ```outlvl=idaeslog.DEBUG```